wpilibc/src/main/native/cpp/smartdashboard/Mechanism2d.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#pragma once
+
+#include "frc/smartdashboard/Mechanism2d.h"
+
+#include <ios>
+
+#include <wpi/raw_ostream.h>
+
+#include "frc/smartdashboard/SendableBuilder.h"
+
+using namespace frc;
+
+Mechanism2d::Mechanism2d(double width, double height) : m_dims{width, height} {}
+
+MechanismRoot2d* Mechanism2d::GetRoot(const wpi::StringRef name, double x, double y) {
+  if (m_roots.count(name)) {
+    return m_roots[name].get();
+  }
+  const auto& twine = wpi::Twine::Twine(name);
+  const std::unique_ptr<MechanismRoot2d> ptr = std::make_unique<MechanismRoot2d>(twine, x, y);
```

Comment:
It's unsafe to store a reference to a temporary.  If you need to wrap it in a Twine for make_unique to be happy, do it in the call (e.g. `std::make_unique<...>(wpi::Twine(name), x, y)`).  It should do an automatic conversion for you though.

Reply:
Don't make the return value const.